pytorch_pfn_extras/training/_trainer.py

```diff
@@ -218,6 +218,8 @@ def run(self,
         self._train_len = train_len
         self._eval_len = eval_len
 
+        device = self.handler._entry_runtime.device_spec  # type: ignore[attr-defined]
```

Comment:
wonder if we should pass the device spec as a trainer arg instead of accessing it like this. It would be much simpler.